### PR TITLE
Rearch video

### PR DESCRIPTION
**Pull request by SPRESENSE**
## Summary
Rearchtect video driver.
- Define two video I/F(struct imgsensor_ops_s and struct imgdata_ops_s),
  and support them.
- Move isx012 driver to general driver directory
- Replace error output macro to verr
- Refactor camera initialization
- Update defconfig of camera example

## Impact
video, isx012 on Spresense board

## Testing
Tested by spresense:example_camera with https://github.com/apache/incubator-nuttx-apps/pull/851

### CODE REVIEW

**Comment by jerpelea** [comment]
@SPRESENSE /github/workspace/sources/nuttx/arch/arm/src/chip/cxd56_cisif.c:1080: undefined reference to `imgdata_register'

**Review by jerpelea — APPROVED** [review]

File: drivers/video/isx012.c [hunk 1]
```diff
@@ -34,12 +34,12 @@
 #include <nuttx/fs/fs.h>
 #include <nuttx/i2c/i2c_master.h>
 #include <arch/board/board.h>
-#include <arch/chip/cisif.h>
+#include <arch/chip/cxd56_cisif.h>
```
Comment (xiaoxiang781216) [inline drivers/video/isx012.c]: it's specific to cxd56

**Comment by SPRESENSE** [comment]
> @SPRESENSE /github/workspace/sources/nuttx/arch/arm/src/chip/cxd56_cisif.c:1080: undefined reference to `imgdata_register'

Fixed a compile error by mpy/defconfig.
There are still build errors, but these will be fixed by the PR https://github.com/apache/incubator-nuttx-apps/pull/851.

**Comment by jerpelea** [comment]
@SPRESENSE drivers/video/isx012.c  can not contain board definitions
#include <arch/chip/cxd56_cisif.h>

please fix it

**Comment by SPRESENSE** [comment]
@jerpelea
In the final code (not in the middle of commit), `drivers/video/isx012.c` does not include `cxd56` and does not depend on `cxd56`. Please check it. Thanks.

**Comment by jerpelea** [comment]
@SPRESENSE please fix https://github.com/apache/incubator-nuttx/pull/4526/commits/f7cfbc11d1fcf0f40d0f4bccb9096b1dcc2a7a48
 so that we have a clean build at every commit and then forcepush

**Pull request merged**